.github/scripts/check_content.py

```diff
@@ -0,0 +1,57 @@
+import os.path
+from os import walk
+
+
+def check_content() -> int:
+    """
+    Walks through the filetree rooted at the current working directory.
+    For each file that it finds, it validates the file
+    """
+    error_count = 0
+    DIRECTORIES_TO_EXCLUDE = ['.git', '.github', '.idea', 'venv', 'DO NOT COMMIT']
+    FILES_TO_EXCLUDE = ['.DS_Store', '.gitignore']
+
+    for root, dirs, files in walk('.', topdown=True):
+        dirs[:] = [d for d in dirs if d not in DIRECTORIES_TO_EXCLUDE]
+        files[:] = [f for f in files if f not in FILES_TO_EXCLUDE]
+        for file in files:
+            relative_path = os.path.join(root, file)
+            error_count += check_file(relative_path, file)
+    return error_count
+
+
+def check_file(relative_path: str, file: str) -> int:
+    """
+    Check the given file for any issues.
+
+    Writes error messages to the console, and returns the number of errors found
+
+    :param relative_path: The path of the file, including the directory, relative to the vault's root
+    :param file: The name of the file, without any directory
+    :return: The number of errors found in the given file 
+    """
+    errors = 0
+
+    if '.' not in file:
+        print(f'Error:\n  This file has no extension: consider adding ".md" to its name:\n  {relative_path} ')
+        errors += 1
+
+    # Other checks may be added here in future
+    return errors
+
+
+def get_root_of_vault() -> str:
+    dir_path = os.path.dirname(os.path.realpath(__file__))
+    up1 = os.path.dirname(dir_path)
+    up2 = os.path.dirname(up1)
+    return up2
```

Comment:
I would add a comment (either # ... or in docstring) that this only works for files in the script category. If ever in the future we have `.github/scripts/script_type/some_script.py`, this no longer works as expected.

Reply:
I must admit that I had assumed that all that mattered was the relative location from the source of this function... so it wouldn't matter if we called it from other levels.

To test this assumption, I just added a test for this function, which runs on my machine and CI, so it looks to me like we are OK on this... 

Or have I missed something? :-)

The test:
https://github.com/claremacrae/obsidian-hub/blob/22c2db7ca776b74ce4c360bc9dacbc81c893e924/.github/scripts/tests/test_check_content.py#L6-L9

The test run:
https://github.com/claremacrae/obsidian-hub/runs/4750104035?check_suite_focus=true#step:5:13